### PR TITLE
feat(components): refactor AutoCompleteTags

### PR DESCRIPTION
**Pull request by marielakas**
Addresses: https://github.com/sumup-oss/circuit-ui/issues/545

## Purpose

When I started working on adding a custom clear icon in the `SearchInput` I went through all components in the `ze-dashboard` project which depend on it. I noticed that we have a `AutoCompleteTags` in our shared components folder which uses the `SearchInput` indirectly and it contained a comment that it should be moved to CircuitUI. However, there were differences in the api of the `AutoCompleteTags` in the dashboard and the component we have here in CircuitUI. 

## Approach and changes

In order to address the aforementioned I made the following changes:

- added `selectedTags` property
- passed `inputProps` to the `AutoCompleteInput`
- fixed search input left icon alignment on Safari

## Definition of done

* [x] Development completed
* [ ] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: src/components/AutoCompleteTags/AutoCompleteTags.js [hunk 1]
```diff
@@ -13,74 +13,95 @@
...
 
-  handleAddTag = option =>
-    this.setState(({ tags }) => ({ tags: [...tags, option] }));
+const AutoCompleteTags = ({
+  availableTags,
+  selectedTags,
+  onChange,
+  ...inputProps
+}) => {
+  const [selected, setSelected] = useState(selectedTags);
```
Comment (connor-baer) [inline src/components/AutoCompleteTags/AutoCompleteTags.js:49]: Head's up! Circuit UI v1 uses an outdated version of React that does not support hooks. If you intend to merge and use this before v2 is released, you'll need to remove any usage of hooks.

**Comment by connor-baer** [comment]
@marielakas Is this PR ready for review? If so, it's missing a proper PR description :)

File: src/components/AutoCompleteTags/AutoCompleteTags.spec.js [hunk 2]
```diff
@@ -14,15 +14,49 @@
...
     const actual = create(<AutoCompleteTags availableTags={[]} />);
     expect(actual).toMatchSnapshot();
   });
+
+  it('should display selected tags ', () => {
+    const { getByTestId, findByText } = render(
+      <AutoCompleteTags {...defaultProps} />
+    );
+    expect(getByTestId('autocomplete-tags-selected')).not.toBeNull();
+    expect(findByText('test1@sumup.com')).not.toBeNull();
```
Comment (connor-baer) [inline src/components/AutoCompleteTags/AutoCompleteTags.spec.js:48]: I'd be more specific here :)
```suggestion
    expect(getByTestId('autocomplete-tags-selected')).toBeVisible();
    expect(findByText('test1@sumup.com')).toBeVisible();
```

File: src/components/Input/Input.js [hunk 3]
```diff
@@ -80,6 +80,7 @@ const inputBaseStyles = ({ theme }) => css`
   transition: border-color ${theme.transitions.default};
   width: 100%;
   ${textMega({ theme })};
+  margin: 0;
```
Comment (connor-baer) [inline src/components/Input/Input.js:83]: This is a breaking change, isn't it?

**Comment by connor-baer** [comment]
In version 2, I'd like to turn this into a stateless, controlled component to give the users greater control over how the state is updated.

File: src/components/AutoCompleteTags/AutoCompleteTags.spec.js [hunk 4]
```diff
@@ -14,15 +14,49 @@
...
+  onChange: jest.fn()
+};
+
 describe('AutoCompleteTags', () => {
+  afterEach(cleanup);
   /**
    * Style tests.
    */
-  it('should render with default styles', () => {
+  it.skip('should render with default styles', () => {
```
[the remark this answers is not in the document]
Comment (marielakas) [reply src/components/AutoCompleteTags/AutoCompleteTags.spec.js:38]: Ooopps :D I forgot about that. I will fix it.

File: src/components/Input/Input.js (hunk 3 above)
Comment (marielakas) [reply src/components/Input/Input.js:83]: Nope, it's a fixing change :D

**Comment by marielakas** [comment]
Makes sense, we could add it to the roadmap for v2. :)

File: src/components/Input/Input.js (hunk 3 above)
Comment (connor-baer) [reply src/components/Input/Input.js:83]: Ah, I confused this with the margin on the outer container. All good 👍

**Pull request merged**